src/linker/rule.py

```diff
@@ -62,23 +64,40 @@ class ImplementedRule(Rule):
     execution_input: List[str]
     validation: str
     output: List[str]
+    resources: Optional[dict]
     envvars: dict
     diagnostics_dir: str
     image_path: str
     script_cmd: str
 
     def _build_rule(self) -> str:
-        return (
-            f"""
+        return self._build_io() + self._build_resources() + self._build_shell_command()
+
+    def _build_io(self) -> str:
+        return f"""
 rule:
     name: "{self.name}"
+    message: "Running Implementation {self.name}"
     input: 
         implementation_inputs={self.execution_input},
         validation="{self.validation}"           
     output: {self.output}
+    log: "{self.diagnostics_dir}/{self.name}-output.log"
     container: "{self.image_path}" """
-            + self._build_shell_command()
-        )
+
+    def _build_resources(self) -> str:
+        # Include slurm partition for now, as snakemake has trouble
+        # parsing strings with periods like "all.q"
```

Comment:
Not sure I understand this -- do you mean how you had to _quote_ it below?

Reply:
Yes, it is related to this bug:
https://github.com/snakemake/snakemake-executor-plugin-slurm/issues/18

Reply:
Ok, gotcha. I would change the word "include" to "single-quote" in this comment.

Reply:
By "include" i mean that in the absence of the bug you wouldn't need to specify it in the rule definition at all; just put it in the default resources that get passed to the snakemake through the CLI. Thinking about it now though, maybe it actually is good to keep the partition defined on the rule level, in case somebody wanted to e.g. put a particular step to the long queue or something. I'll make the change you suggested.

Reply:
Ah, I see. Yes I think eventually we will want all resources to be defined at the rule level, so this is actually getting a bit ahead of ourselves! 😄